### PR TITLE
Test for derived staking query

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Fixed staking derive queries and added tests.
Closes https://github.com/cennznet/api.js/issues/273

### CODE REVIEW

File: packages/api/src/derives/index.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ import * as session from './session';
 import * as staking from './staking';
 import * as balances from './balances';
 
-export type DeriveFunc = (instanceId: string, api: ApiInterfaceRx) => (...args: CodecArg[]) => Observable<any>;
+export type DeriveFunc = (instanceId: string, api: ApiInterfaceRx) => (...args: any[]) => Observable<any>;
```
Comment (KarishmaBothara) [inline packages/api/src/derives/index.ts:27]: The arguments can't be of type CodeArg always...
https://github.com/KarishmaBothara/api/blob/772769c981dbf7d7e051c59fbd91b1bc9bb67312/packages/api-derive/src/index.ts#L42

File: packages/api/test/e2e/staking.e2e.ts [hunk 2]
```diff
@@ -33,6 +33,59 @@ afterAll(async () => {
   await api.disconnect();
 });
 
+describe('Staking derived queries', () => {
+
+  it('test elected validators info for local chain', async done =>{
```
Comment (holygits) [inline packages/api/test/e2e/staking.e2e.ts:38]:
```suggestion
  test('elected validators info for local chain', async done =>{
```

File: packages/api/test/e2e/staking.e2e.ts [hunk 3]
```diff
@@ -33,6 +33,59 @@ afterAll(async () => {
...
+    expect(info[0].exposure.others.length).toEqual(0);
+    expect(info[0].nominators.length).toEqual(0);
+    expect(info[0].rewardDestination.toString()).toEqual('Stash');
+    expect(info[0].stashId.toString()).toEqual('5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY');
+    expect(nextElected.length).toBeGreaterThanOrEqual(1);
+    expect(validators.length).toBeGreaterThanOrEqual(1);
+    done();
+  })
+
+  it('test waiting validators info query', async done =>{
```
Comment (holygits) [inline packages/api/test/e2e/staking.e2e.ts:55]:
```suggestion
  test('waiting validators info query', async done =>{
```

File: packages/api/test/e2e/staking.e2e.ts [hunk 4]
```diff
@@ -33,6 +33,59 @@ afterAll(async () => {
...
+    const waitingValidatorsToBeElected = await api.derive.staking.waitingInfo();
+    const {info, waiting} = waitingValidatorsToBeElected;
+    expect(info.length).toEqual(0);
+    expect(waiting.length).toEqual(0);
+    done();
+  })
+
+  it('validators info', async done => {
```
Comment (holygits) [inline packages/api/test/e2e/staking.e2e.ts:63]:
```suggestion
  test('validators info', async done => {
```

File: packages/api/test/e2e/staking.e2e.ts [hunk 5]
```diff
@@ -33,6 +33,59 @@ afterAll(async () => {
...
+    done();
+  })
+
+  it('get all validators', async done => {
+    const validatorList = await api.derive.staking.stashes();
+    expect(validatorList[0].toString()).toEqual('5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY');
+    done();
+  })
+
+  it('test staking overview query', async done => {
```
Comment (holygits) [inline packages/api/test/e2e/staking.e2e.ts:76]:
```suggestion
  test('staking overview query', async done => {
```

**Review by holygits — APPROVED** [review]
lgtm

**Pull request merged**